src/main/java/fr/insalyon/creatis/gasw/parser/GaswParser.java

```diff
@@ -489,4 +489,89 @@ private static void addName(
             content.append(separator).append(f.getName());
         }
     }
-}
+
+    public GaswInput getGaswInput(String applicationName, Map<String, String> inputsMap, String boutiquesFilePath, String executableName,HashMap<Integer, String> inputid , HashMap<Integer, String> outputid, String invocationString, 
+    Map<String, String> resultDirectory, String jobId, String sourceFilePath, List<URI> DownloadFiles, String outputDirName)
+            throws URISyntaxException, FileNotFoundException, IOException, GaswException, ParseException, SAXException {
```

Comment:
This is almost the same as the original getGaswInput method, we should try to use the same logic for both.

Reply:
tried to simplify the logic according to the original getGaswInput